pkg/util/featuregates.go

```diff
@@ -0,0 +1,56 @@
+package util
+
+import (
+	"strings"
+
+	"k8s.io/klog/v2"
+
+	v1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver/featuregates"
+	upstreamfeature "k8s.io/component-base/featuregate"
+	cloudfeatures "k8s.io/controller-manager/pkg/features"
+)
+
+// GetEnabledDisabledFeatures returns two slices that contain all the known feature gates
+// and separates them by their enabled/disabled state. It filters features so that we return
+// only those that are to be consumed by the cloud provider instead of all that exist
+// in the kubernetes ecosystem.
+func GetEnabledDisabledFeatures(features featuregates.FeatureGate) ([]string, []string) {
+	var enabled []string
+	var disabled []string
+
+	upstreamFeatureGate := upstreamfeature.NewFeatureGate()
+	if err := cloudfeatures.SetupCurrentKubernetesSpecificFeatureGates(upstreamFeatureGate); err != nil {
+		klog.Errorf("can not get feature gates: %v", err)
+		return nil, nil
+	}
```

Comment:
This should return an error and block when it can't get the feature gates, we shouldn't proceed in this case

Reply:
ACK, will do